src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.impl.request.ShoppingCartRequestMapper;
+import mate.academy.spring.mapper.impl.response.OrderResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.request.ShoppingCartRequestDto;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private OrderResponseMapper orderResponseMapper;
+    private OrderService orderService;
+    private ShoppingCartRequestMapper shoppingCartRequestMapper;
+    private UserService userService;
+
+    public OrderController(OrderResponseMapper orderResponseMapper,
+                           OrderService orderService,
+                           ShoppingCartRequestMapper shoppingCartRequestMapper,
+                           UserService userService) {
+        this.orderResponseMapper = orderResponseMapper;
+        this.orderService = orderService;
+        this.shoppingCartRequestMapper = shoppingCartRequestMapper;
+        this.userService = userService;
+    }
+
+    @PostMapping("/complete/{userId}")
+    public OrderResponseDto complete(@PathVariable Long userId,
+                                          @RequestBody ShoppingCartRequestDto
+                                                  shoppingCartRequestDto) {
+        Order completeOrder = orderService.completeOrder(shoppingCartRequestMapper
+                .fromDto(shoppingCartRequestDto));
+        return orderResponseMapper.toDto(completeOrder);
+    }
+
+    @GetMapping("/{userId}")
```

Comment:
should be a request param
user id as a path variable is usually applicable in UserController
order id in OrderController